Dockerfile

```
@@ -0,0 +1,38 @@
+
+# Cache dependency compilation separately from source changes.
+COPY Cargo.toml Cargo.lock ./
+RUN mkdir src && echo 'fn main(){}' > src/main.rs \
+    && cargo build --release 2>/dev/null; rm -rf src
+
+COPY src ./src
+RUN touch src/main.rs && cargo build --release
+
+# Stage 3: minimal runtime image
+FROM debian:bookworm-slim
+RUN apt-get update && apt-get install -y \
```

Hadolint (2.14.0)

[info] Avoid additional packages by specifying `--no-install-recommends`

(DL3015)